mbuild/path.py

```diff
@@ -13,7 +15,8 @@
 
 
 class Path:
-    def __init__(self, N=None, coordinates=None, bond_graph=None):
+    def __init__(self, N=None, coordinates=None, bond_graph=None, path_graph=None):
```

Comment:
I'm kind of confused about the difference between `bond_graph` and `path_graph` here. With the current behavior, you pass in a networkx graph, and that is used to create the bond graph when converting the path to a compound with the `set_bond_graph` method in Compound.

So, a linear path graph is achieved by passing in:
`bond_graph = nx.path_graph(N)`

But..now as I'm typing this up, I realize that one of the issues with this is that you have to predefine N. This works for random walk paths, but not ones where you don't know N initially (Lamellar, Circular in certain cases).  Does the "linear" graph solve this?